### PR TITLE
Log Version

### PR DESCRIPTION
**Pull request by sowu880**

### CODE REVIEW

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/JobManager.cs [hunk 1]
```diff
@@ -48,6 +49,10 @@ public class JobManager
         public async Task RunAsync(CancellationToken cancellationToken = default)
         {
             _logger.LogInformation("Job manager starts running.");
+            var assembly = Assembly.GetExecutingAssembly();
+            var version1 = assembly.GetName().Version;
+            var version = ((AssemblyFileVersionAttribute)Attribute.GetCustomAttribute(assembly, typeof(AssemblyFileVersionAttribute), false)).Version;
+            _logger.LogInformation($"Running version:{version1} FHIR to Synapse Pipeline. Fileversion: {version}.");
```
Comment (moria97) [inline FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/JobManager.cs:55]: Can we check whether the correct version can be logged in the image?

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Tool/Dockerfile [hunk 2]
```diff
@@ -18,7 +18,7 @@ RUN dotnet restore "src/Microsoft.Health.Fhir.Synapse.Tool/Microsoft.Health.Fhir
 
 COPY . .
 WORKDIR "/src/src/Microsoft.Health.Fhir.Synapse.Tool"
-RUN dotnet build "Microsoft.Health.Fhir.Synapse.Tool.csproj" -c Release -o /app/build
+RUN dotnet build "Microsoft.Health.Fhir.Synapse.Tool.csproj" -c Release -o /app/build -p:Version=0.5.0
 FROM build AS publish
 
 RUN dotnet publish "Microsoft.Health.Fhir.Synapse.Tool.csproj" -c Release -o /app/publish /p:UseAppHost=false
```
Comment (moria97) [inline FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Tool/Dockerfile:24]: do we need version in publish command?

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/JobManager.cs [hunk 3]
```diff
@@ -49,6 +50,9 @@ public async Task RunAsync(CancellationToken cancellationToken = default)
         {
             _logger.LogInformation("Job manager starts running.");
```
Comment (moria97) [inline FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/JobManager.cs:51]: We can just log $"Job manager starts running with version {}"

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Tool/Dockerfile [hunk 4]
```diff
@@ -18,7 +18,7 @@ RUN dotnet restore "src/Microsoft.Health.Fhir.Synapse.Tool/Microsoft.Health.Fhir
 
 COPY . .
 WORKDIR "/src/src/Microsoft.Health.Fhir.Synapse.Tool"
-RUN dotnet build "Microsoft.Health.Fhir.Synapse.Tool.csproj" -c Release -o /app/build
+RUN dotnet build "Microsoft.Health.Fhir.Synapse.Tool.csproj" -c Release -o /app/build -p:Version=0.5.0
```
Comment (QuanWanxx) [inline FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Tool/Dockerfile:21]: Not in this PR, I feel we can use a static place for pipeline version, currently we have version value in 1. ARM Template. 2. Powershell Script. 3. CI Yaml. 4. This Dockerfile

Comment (moria97) [reply FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Tool/Dockerfile:21]: +1 let's file an ADO item to track this in github issues

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Tool/Dockerfile (hunk 2 above)
Comment (sowu880) [reply FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Tool/Dockerfile]: Yes, updated it.

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/JobManager.cs (hunk 3 above)
Comment (sowu880) [reply FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/JobManager.cs]: Updated.

File: FhirToDataLake/azurepipelines/ci.yml [hunk 5]
```diff
@@ -110,7 +110,7 @@ steps:
     targetType: inline
     workingDirectory: FhirToDataLake
     script: |
-      docker build -t $(imageName):$(version) -f ./src/Microsoft.Health.Fhir.Synapse.Tool/Dockerfile .
+      docker build -t $(imageName):$(version) -f ./src/Microsoft.Health.Fhir.Synapse.Tool/Dockerfile . --build-arg Version=$(version)
```
Comment (moria97) [inline FhirToDataLake/azurepipelines/ci.yml:113]: 👍

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/JobManager.cs [hunk 6]
```diff
@@ -47,7 +48,8 @@ public class JobManager
         /// <returns>Completed task.</returns>
         public async Task RunAsync(CancellationToken cancellationToken = default)
         {
-            _logger.LogInformation("Job manager starts running.");
+            var assembly = Assembly.GetExecutingAssembly();
+            _logger.LogInformation($"Job manager starts running with version {assembly.GetName().Version}");
```
Comment (moria97) [inline FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/JobManager.cs:52]: Can we merge into one line?

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.FunctionApp/Program.cs [hunk 7]
```diff
@@ -22,6 +25,9 @@ public class Program
     {
         public static void Main()
         {
+            var assembly = Assembly.GetExecutingAssembly();
```
Comment (moria97) [inline FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.FunctionApp/Program.cs:28]: We can remove these two lines here

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Tool/Program.cs [hunk 8]
```diff
@@ -23,6 +25,8 @@ public class Program
     {
         public static async Task Main(string[] args)
         {
+            var assembly = Assembly.GetExecutingAssembly();
```
Comment (moria97) [inline FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Tool/Program.cs:28]: We can remove these two lines as log has already been written in JobManager

File: FhirToDataLake/azurepipelines/ci.yml (hunk 5 above)
Comment (moria97) [reply FhirToDataLake/azurepipelines/ci.yml:113]: Maybe we can use another parameter like `binaryversion` or `assemblyversion` to avoid ambiguity

**Review by moria97 — APPROVED** [review]

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Tool/Program.cs (hunk 8 above)
Comment (sowu880) [reply FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Tool/Program.cs]: Updated.

**Pull request merged**